Resources/public/js/views/ez-usermenuview.js

```diff
@@ -0,0 +1,243 @@
+            var methodName = this.get('displayed') ? 'addClass' : 'removeClass',
+                userMenu = this.get('container').one(SELECTOR_USER_MENU);
+
+            userMenu[methodName](CLASS_HIDDEN);
+            this.set('displayed', !this.get('displayed'));
+        },
+
+        /**
+         * Checks if clicked outside the user profile and hide the user menu if is vissible. 
+         *
+         * @method _checkClickedOutside
+         * @protected
+         */
+        _checkClickedOutside: function (event) {
+            if (!event.target.hasClass(CLASS_USER_PROFILE) && event.target.ancestor(SELECTOR_USER_PROFILE) === null && this.get('displayed')) {
+                 this._toggleUserMenu();
+            }
+        },
+
+        /**
+         * Renders views for the menu items
+         *
+         * @method _renderItemsView
+         * @protected
+         */
+        _renderItemsView: function () {
+            var container = this.get('container'),
+                view;
+
+            Y.Object.each(this.get('items'), Y.bind(function (item, name) {
+                view = this.get(name + 'ItemView');
+
+                view.addTarget(this);
+                container.one(SELECTOR_USER_MENU).append(view.render().get('container'));
+                view.set('active', true);
+            }, this));
+        },
+
+        /**
+         * Returns a user menu item object
+         *
+         * @method _getItem
+         * @protected
+         * @param title {String} the title to show in user menu
+         * @param classSuffix {String} the class suffix
+         */
+        _getItem: function (title, classSuffix) {
+            return {
+                Constructor: Y.eZ.UserMenuItemView,
+                config: {
+                    title: title,
+                    classSuffix: classSuffix
+                }
+            };
+        },
+
+        /**
+         * Builds the user menu item view based on the value.
+         *
+         * @method _buildItemView
+         * @protected
+         * @param item {Object} plain object or Y.View
+         * @return {Object} the Y.View
+         */
+        _buildItemView: function (item) {
+            var ViewConstructor,
+                view;
+
+            if (item instanceof Y.eZ.UserMenuItemView) {
+                view = item;
+            } else {
+                ViewConstructor = item.Constructor;
+                view = new ViewConstructor(item.config || {});
+            }
+
+            view.addTarget(this);
+
+            return view;
+        },
+    }, {
+        ATTRS: {
+            /**
+             * Contains the title and class suffix for the user menu items
+             *
+             * @attribute items
+             * @default {
+             *     notification: 'notification',
+             *     logOut: 'logOut'
+             * }
+             * @type {Object}
+             */
+            items: {
+                value: {
```

Comment:
I think it would be better to use an array here instead of object.
You can easily sort an array.